`TypeError: module.require is not a function` error when running with turbopack

### Issue Description

Apologies if this should be an FR, or an issue on a different project. I'm finding it hard to work out if nextjs + turbopack + apollo server is supposed to be supported or not, and what's stopping it from working at the moment.

In any case, when trying to run `next dev --turbo` on a project with apollo server, I get a bunch of `TypeError: module.require is not a function` errors whenever the gql endpoint is hit, and no data is returned. Running without `--turbo` (i.e. using webpack) works fine.

Related issues:
https://github.com/apollo-server-integrations/apollo-server-integration-next/issues/172
https://github.com/apollo-server-integrations/apollo-server-integration-next/issues/232
https://github.com/vercel/next.js/issues/72573

Thanks

### Link to Reproduction

See https://github.com/vercel/next.js/issues/72573

### Reproduction Steps

See https://github.com/vercel/next.js/issues/72573

I wanted to look into this, but when I checked out @MersadHabibi's linked repo and ran `npm i` I got this:

```
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: react-hook-form@7.53.2
npm error Found: react@19.0.0-rc-66855b96-20241106
npm error node_modules/react
npm error   react@"19.0.0-rc-66855b96-20241106" from the root project
npm error   peerOptional react@"^16.8.0 || ^17.0.0 || ^18.0.0 || >=19.0.0-rc <19.0.0" from @apollo/client@3.11.9
npm error   node_modules/@apollo/client
npm error     @apollo/client@"^3.11.9" from the root project
npm error   5 more (lucide-react, next, react-dom, rehackt, styled-jsx)
npm error
npm error Could not resolve dependency:
npm error peer react@"^16.8.0 || ^17 || ^18 || ^19" from react-hook-form@7.53.2
npm error node_modules/react-hook-form
npm error   react-hook-form@"^7.53.2" from the root project
npm error   peer react-hook-form@"^7.0.0" from @hookform/resolvers@3.9.1
npm error   node_modules/@hookform/resolvers
npm error     @hookform/resolvers@"^3.9.1" from the root project
npm error
npm error Conflicting peer dependency: react@19.0.0
npm error node_modules/react
npm error   peer react@"^16.8.0 || ^17 || ^18 || ^19" from react-hook-form@7.53.2
npm error   node_modules/react-hook-form
npm error     react-hook-form@"^7.53.2" from the root project
npm error     peer react-hook-form@"^7.0.0" from @hookform/resolvers@3.9.1
npm error     node_modules/@hookform/resolvers
npm error       @hookform/resolvers@"^3.9.1" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

I'd be happy to look in more detail given a full detailed reproduction that requires no creativity on my part. Though it does seem likely that this is just some overzealous optimization on "turbopack"'s part?

> I wanted to look into this, but when I checked out @MersadHabibi's linked repo and ran `npm i` I got this:
> 
> ```
> npm error code ERESOLVE
> npm error ERESOLVE could not resolve
> npm error
> npm error While resolving: react-hook-form@7.53.2
> npm error Found: react@19.0.0-rc-66855b96-20241106
> npm error node_modules/react
> npm error   react@"19.0.0-rc-66855b96-20241106" from the root project
> npm error   peerOptional react@"^16.8.0 || ^17.0.0 || ^18.0.0 || >=19.0.0-rc <19.0.0" from @apollo/client@3.11.9
> npm error   node_modules/@apollo/client
> npm error     @apollo/client@"^3.11.9" from the root project
> npm error   5 more (lucide-react, next, react-dom, rehackt, styled-jsx)
> npm error
> npm error Could not resolve dependency:
> npm error peer react@"^16.8.0 || ^17 || ^18 || ^19" from react-hook-form@7.53.2
> npm error node_modules/react-hook-form
> npm error   react-hook-form@"^7.53.2" from the root project
> npm error   peer react-hook-form@"^7.0.0" from @hookform/resolvers@3.9.1
> npm error   node_modules/@hookform/resolvers
> npm error     @hookform/resolvers@"^3.9.1" from the root project
> npm error
> npm error Conflicting peer dependency: react@19.0.0
> npm error node_modules/react
> npm error   peer react@"^16.8.0 || ^17 || ^18 || ^19" from react-hook-form@7.53.2
> npm error   node_modules/react-hook-form
> npm error     react-hook-form@"^7.53.2" from the root project
> npm error     peer react-hook-form@"^7.0.0" from @hookform/resolvers@3.9.1
> npm error     node_modules/@hookform/resolvers
> npm error       @hookform/resolvers@"^3.9.1" from the root project
> npm error
> npm error Fix the upstream dependency conflict, or retry
> npm error this command with --force or --legacy-peer-deps
> npm error to accept an incorrect (and potentially broken) dependency resolution.
> ```
> 
> I'd be happy to look in more detail given a full detailed reproduction that requires no creativity on my part. Though it does seem likely that this is just some overzealous optimization on "turbopack"'s part?

use `npm i --legacy-peer-deps` or `npm i --force`

--- a/.changeset/khaki-socks-obey.md
+++ b/.changeset/khaki-socks-obey.md
@@ -1,0 +1,7 @@
+---
+'@apollo/server-integration-testsuite': patch
+'@apollo/server-plugin-response-cache': patch
+'@apollo/server': patch
+---
+
+Compatibility with Next.js Turbopack. Fixes #8004.

--- a/cspell-dict.txt
+++ b/cspell-dict.txt
@@ -190,6 +190,7 @@ triaging
 tsbuildinfo
 tsconfig
 tsconfigs
+Turbopack
 typecheck
 typeis
 typenames

--- a/package.json
+++ b/package.json
@@ -41,7 +41,7 @@
     "@apollo/client": "3.12.4",
     "@apollo/gateway": "2.9.3",
     "@apollo/subgraph": "2.9.3",
-    "@apollo/utils.createhash": "2.0.1",
+    "@apollo/utils.createhash": "2.0.2",
     "@changesets/changelog-github": "0.5.0",
     "@changesets/cli": "2.27.11",
     "@graphql-codegen/cli": "3.3.1",

--- a/packages/integration-testsuite/package.json
+++ b/packages/integration-testsuite/package.json
@@ -30,7 +30,7 @@
     "@apollo/client": "^3.6.9",
     "@apollo/server": "4.11.2",
     "@apollo/usage-reporting-protobuf": "^4.1.1",
-    "@apollo/utils.createhash": "^2.0.0",
+    "@apollo/utils.createhash": "^2.0.2",
     "@apollo/utils.keyvaluecache": "^2.1.0",
     "express": "^4.21.1",
     "graphql-http": "1.22.3",

--- a/packages/plugin-response-cache/package.json
+++ b/packages/plugin-response-cache/package.json
@@ -32,7 +32,7 @@
     "node": ">=14.16.0"
   },
   "dependencies": {
-    "@apollo/utils.createhash": "^2.0.0",
+    "@apollo/utils.createhash": "^2.0.2",
     "@apollo/utils.keyvaluecache": "^2.1.0"
   },
   "peerDependencies": {

--- a/packages/server/package.json
+++ b/packages/server/package.json
@@ -136,7 +136,7 @@
     "@apollo/cache-control-types": "^1.0.3",
     "@apollo/server-gateway-interface": "^1.1.1",
     "@apollo/usage-reporting-protobuf": "^4.1.1",
-    "@apollo/utils.createhash": "^2.0.0",
+    "@apollo/utils.createhash": "^2.0.2",
     "@apollo/utils.fetcher": "^2.0.0",
     "@apollo/utils.isnodelike": "^2.0.0",
     "@apollo/utils.keyvaluecache": "^2.1.0",

